File: app/src/main/java/com/warunya/ricenearby/constant/Gender.java
package com.warunya.ricenearby.constant;

import com.google.gson.annotations.SerializedName;

import java.util.HashMap;

public enum Gender {

    @SerializedName("0")
    None,
    @SerializedName("1")
    Male,
    @SerializedName("2")
    Female;

    public static Gender parse(int type) {
        HashMap<Integer, Gender> creatorMap = new HashMap<>();
        creatorMap.put(None.ordinal(), None);
        creatorMap.put(Male.ordinal(), Male);
        creatorMap.put(Female.ordinal(), Female);

        return (creatorMap.get(type) == null) ? None : creatorMap.get(type);
    }
}

File: app/src/main/java/com/warunya/ricenearby/constant/Gender.kt
package com.warunya.ricenearby.constant

import com.google.gson.annotations.SerializedName
import java.util.*

enum class Gender {

    @SerializedName("0")
    None,
    @SerializedName("1")
    Male,
    @SerializedName("2")
    Female;

    companion object {
        fun parse(type: Int): Gender {
            val creatorMap = HashMap<Int, Gender>()
            creatorMap[Male.ordinal] = Male
            creatorMap[Female.ordinal] = Female
            return if (creatorMap[type] == null) None else creatorMap[type]!!
        }
    }
}